diff --git a/distrho/src/DistrhoPluginLV2export.cpp b/distrho/src/DistrhoPluginLV2export.cpp
index 916b86f2..59fa85c6 100644
--- a/distrho/src/DistrhoPluginLV2export.cpp
+++ b/distrho/src/DistrhoPluginLV2export.cpp
@@ -1176,16 +1176,11 @@ void lv2_generate_ttl(const char* const basename)
         {
             const uint32_t version(plugin.getVersion());
 
-            const uint32_t majorVersion = (version & 0xFF0000) >> 16;
-            /* */ uint32_t minorVersion = (version & 0x00FF00) >> 8;
+            const uint32_t minorVersion = (version & 0x00FF00) >> 8;
             const uint32_t microVersion = (version & 0x0000FF) >> 0;
 
-            // NOTE: LV2 ignores 'major' version and says 0 for minor is pre-release/unstable.
-            if (majorVersion > 0)
-                minorVersion += 2;
-
-            pluginString += "    lv2:microVersion " + String(microVersion) + " ;\n";
-            pluginString += "    lv2:minorVersion " + String(minorVersion) + " .\n";
+            pluginString += "    lv2:minorVersion " + String(minorVersion) + " ;\n";
+            pluginString += "    lv2:microVersion " + String(microVersion) + " .\n";
         }
 
         // port groups
@@ -1713,3 +1708,4 @@ void lv2_generate_ttl(const char* const basename)
     }
 #endif
 }
+
